### PR TITLE
Use main ci for building docker images

### PR DESCRIPTION
**Pull request by mdesouky**
### Changes:

- Configure test and publish workflows to use our default `ci` reusable github actions workflow

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -15,38 +15,24 @@ permissions:
 env:
   MISE_PYTHON_COMPILE: false
   GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  AWS_IAM_ROLE: ""
```
Comment (uptickmetachu) [inline .github/workflows/publish.yml:18]: This still needed?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -24,3 +24,19 @@ jobs:
       command: |
         mise run install
         mise run ci
+  build:
```
Comment (uptickmetachu) [inline .github/workflows/test.yml:27]: We can remove this now no?

**Review by uptickmetachu — DISMISSED** [review]

File: .github/workflows/publish.yml (hunk 1 above)
Comment (mdesouky) [reply .github/workflows/publish.yml]: removed 👍🏼

File: .github/workflows/test.yml (hunk 2 above)
Comment (mdesouky) [reply .github/workflows/test.yml:27]: @uptickmetachu we don't want test builds ?

Comment (uptickmetachu) [reply .github/workflows/test.yml:27]: I guess if we have the lifecycle policies. 

I don't think its entirely too neccessary but it doesn't hurt too much.

**Review by uptickmetachu — APPROVED** [review]

**Pull request merged**